### PR TITLE
feat: add base components, styling, auth hooks for frontend

### LINKED ISSUE #32: feat: add base components, fonts, colors for frontend

### PR DESCRIPTION
**Pull request by aidanaden**
Closes #32 

Changes include:
- updates backend `/refresh` endpoint to use `GET`
- adds base ui components
- adds frontend font
- adds auth + websocket stores
- adds responsive navbar
- adds login/signup/reset password pages
- adds dashboard page
- completes entire auth flow

### CODE REVIEW

**Comment by blaketingyu** [comment]
Frontend god 💧

File: frontend/src/components/layout/radiogroup/index.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
+import { useState } from "react";
+import { RadioGroup } from "@headlessui/react";
+import cx from "classnames";
+
+import { CheckIcon } from "src/components/icons/CheckIcon";
+
+const difficulties = [
+  {
+    name: "easy",
+    description:
```
Comment (yusufaine) [inline frontend/src/components/layout/radiogroup/index.tsx:10]: descriptions currently are placeholders until we can think of something better to replace it with -- leetcode currently doesnt "describe" what each difficulty level means so we can prolly make our own definition 

* Easy: Everyone starts here.
* Medium: Once you're comfortably answering questions here, you're well on your way to securing a good internship.
* Hard: The world is your oyster -- FAANG, HFT, IB, you name it.

File: frontend/src/components/layout/navbar/index.tsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    <nav className="font-display pl-4 pr-2 py-3 md:p-4 bg-neutral-100 fixed w-full z-50 top-0 left-0">
+      <div className="max-w-5xl flex flex-wrap justify-between items-center mx-auto">
+        <BaseLink href="/" className="flex items-center h-full">
+          {/* <img
+            src="https://flowbite.com/docs/images/logo.svg"
+            className="mr-3 h-6 sm:h-9"
+            alt="Flowbite Logo"
+          /> */}
+          <span className="font-display self-center text-2xl font-semibold whitespace-nowrap text-neutral-800">
+            PussyPrep
```
Comment (yusufaine) [inline frontend/src/components/layout/navbar/index.tsx:95]:
```suggestion
            PeerPrep
```

Just in case

**Review by yusufaine — APPROVED** [review]
Left a comment on the difficulty descriptors at the home page, not really set on the suggestion but can KIV since its not too important.

Also left a change suggestion for the logo name ah HAHAH

Apart from that, tested it locally, I can verify that auth flow works as intended.

File: frontend/src/login/components/ResetPasswordForm.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+      ) : (
+        <h4 className="leading-tight text-1xl text-black-50 flex flex-col text-center mb-4">
+          Please enter your new password
+        </h4>
+      )}
+      <form className="flex flex-col gap-8" onSubmit={onSubmit}>
+        <TextInput
+          label="New Password"
+          type="password"
+          placeholder="Very$ecureP4ssword"
```
Comment (blaketingyu) [inline frontend/src/login/components/ResetPasswordForm.tsx:60]: Are we keeping this as a placeholder?

**Review by blaketingyu — APPROVED** [review]
LGTM!

**Pull request merged**